src/socket/linux.rs

```diff
@@ -109,9 +109,17 @@ impl<A: NetworkAddress, S> Socket<A, S> {
 
 pub(super) fn configure_timestamping(
     socket: &RawSocket,
+    interface: Option<InterfaceName>,
     mode: InterfaceTimestampMode,
-    bind_phc: Option<u32>,
+    mut bind_phc: Option<u32>,
 ) -> std::io::Result<()> {
+    // Check if the phc is not the interface-native phc.
+    if let Some(interface) = interface {
+        if lookup_phc(interface) == bind_phc {
```

Comment:
If bind_phc is already None, do we still need to do the check here?

Reply:
Technically no, but it is cheap and this way is a lot more understandable.